server/lib/chat/chat-service.ts

```diff
@@ -206,12 +232,11 @@ export default class ChatService {
   }
 
   private subscribeUserToChannel(userSockets: UserSocketsGroup, channelName: string) {
-    userSockets.subscribe(getChannelPath(channelName), () => {
-      return {
-        action: 'init',
-        activeUsers: this.state.channels.get(channelName),
-      }
-    })
+    const initEventData: ChatInitEvent = {
+      action: 'init',
+      activeUsers: (this.state.channels.get(channelName) ?? Set()).toArray(),
```

Comment:
nit: prefer creating the JS things as a fallback instead of creating an immutable thing and then converting it, e.g.
```ts
this.state.channels.get(channelName)?.toArray() ?? []
```

Although honestly for this particular case, if this isn't set, this seems like programmer error, so maybe `!` is just a more appropriate thing to do